sdk-lib/src/main/rust/lib.rs

```diff
@@ -1091,18 +1126,46 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_jni_RustBackend_createToAdd
 
         let prover = LocalTxProver::new(Path::new(&spend_params), Path::new(&output_params));
 
-        create_spend_to_address(
-            &mut db_data,
-            &network,
-            prover,
-            &usk,
-            &to,
-            value,
+        let request = TransactionRequest::new(vec![Payment {
+            recipient_address: to,
+            amount: value,
             memo,
-            OvkPolicy::Sender,
-            ANCHOR_OFFSET,
+            label: None,
+            message: None,
+            other_params: vec![],
+        }])
+        .map_err(|e| format_err!("Error creating transaction request: {:?}", e))?;
+
+        zip317_helper(
+            (&mut db_data, prover, request),
+            use_zip317_fees,
+            |(wallet_db, prover, request), input_selector| {
+                spend(
+                    wallet_db,
+                    &network,
+                    prover,
+                    &input_selector,
+                    &usk,
+                    request,
+                    OvkPolicy::Sender,
+                    ANCHOR_OFFSET,
+                )
+                .map_err(|e| format_err!("Error while creating transaction: {}", e))
+            },
+            |(wallet_db, prover, request), input_selector| {
+                spend(
+                    wallet_db,
+                    &network,
+                    prover,
+                    &input_selector,
+                    &usk,
+                    request,
+                    OvkPolicy::Sender,
+                    ANCHOR_OFFSET,
+                )
+                .map_err(|e| format_err!("Error while creating transaction: {}", e))
+            },
```

Comment:
This is an unfortunate duplication. I'm not sure how to fix that given that the types are different.